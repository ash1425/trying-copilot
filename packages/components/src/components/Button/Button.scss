.btn {
  border-radius: 4px;
  color: #fff;
  display: flex;
  flex-direction: column;
  height: 40px;
  justify-content: center;
  margin: 10px;
  padding: 4px;
  text-align: center;

  &.primary {
    background: $primary;
  }

  &.secondary {
    background: $secondary;
  }
}
