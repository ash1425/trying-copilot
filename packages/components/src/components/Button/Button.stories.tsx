import React from "react";

import { Meta, Story } from "@storybook/react";

import Button, { IButtonProps } from "./Button";

export default {
  component: Button,
  title: "components/Button",
  argTypes: {
    className: {
      options: ["primary", "secondary"],
      control: { type: "radio" },
    },
    children: { options: "Button", control: { type: "text" } },
  },
} as Meta;

const Template: Story<IButtonProps> = (args) => {
  const { onClick, className, children } = args;
  return (
    <Button className={className} onClick={onClick}>
      {children}
    </Button>
  );
};
export const Primary = Template.bind({});

Primary.args = { className: "primary", children: "Button" };

export const Secondary = Template.bind({});

Secondary.args = { className: "secondary", children: "Button" };
