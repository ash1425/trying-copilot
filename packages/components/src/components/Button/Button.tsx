import React from "react";
import "./Button.scss";

export interface IButtonProps {
  className: string;
  onClick: Function;
  children: React.ReactNode;
}

const Button = ({ className, onClick, children }: IButtonProps) => (
  <div className={`btn ${className}`} onClick={(e) => onClick(e)}>
    {children}
  </div>
);

export default Button;
