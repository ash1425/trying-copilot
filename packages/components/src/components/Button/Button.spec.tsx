import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./Button";

describe("Button Componentt", () => {
  test("renders Button component", () => {
    render(
      <Button className="dummy-class" onClick={() => {}}>
        Test Label
      </Button>
    );
  });

  test("Verifies the text", () => {
    const { container } = render(
      <Button className="dummy-class" onClick={() => {}}>
        Test Label
      </Button>
    );
    expect(screen.getByText("Test Label")).toHaveTextContent("Test Label");
    expect(container.firstChild).toHaveClass("dummy-class");
  });
  test("Verifies the onclick", () => {
    const onSubmit = jest.fn();
    const { asFragment } = render(
      <Button className="dummy-class" onClick={onSubmit}>
        Test Label
      </Button>
    );
    userEvent.click(screen.getByText("Test Label"));
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(asFragment()).toMatchSnapshot();
  });
});
