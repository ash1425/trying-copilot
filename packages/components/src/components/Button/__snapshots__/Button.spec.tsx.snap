// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Componentt Verifies the onclick 1`] = `
<DocumentFragment>
  <div
    class="btn dummy-class"
  >
    Test Label
  </div>
</DocumentFragment>
`;
